test/browser-specific/html-reporter.spec.js

```diff
@@ -0,0 +1,157 @@
+/* global Mocha */
+'use strict';
+
+describe('HTML reporter', function () {
+  var Suite = Mocha.Suite;
+  var Runner = Mocha.Runner;
+  var HTML = Mocha.reporters.HTML;
+
+  function removeMochaElem () {
+    var mochaElem = document.getElementById('mocha');
+    if (mochaElem) {
+      mochaElem.parentNode.removeChild(mochaElem);
+    }
+  }
+
+  function addMochaElem () {
+    var mochaElem = document.createElement('div');
+    mochaElem.id = 'mocha';
+    document.body.appendChild(mochaElem);
+  }
+
+  function sleep (fn, ms) {
```

Comment:
this in particular is going to be brittle / susceptible to race conditions.  but this test suite is unlike anything else we have, so we don't have the tooling.

all of assertions involving markup in `innerHTML` are also going to be brittle, but again, need better tooling.

I'm going to grab this PR and test it against saucelabs' browsers.  if it is flaky, we'll have to change it.  but if not, we'll just merge and deal w/ it later--unless, of course, you feel like pulling in [Nightwatch](http://nightwatchjs.org).

Reply:
Thank you for reviewing.
Since I think I will not be able to achieve it,
Would you please grab this pull request ?